components/arm/universalrobots/ur.go

```diff
@@ -440,3 +455,80 @@ func reader(ctx context.Context, conn io.Reader, ua *URArm, onHaveData func()) e
 		}
 	}
 }
+
+const errBuiltinKinematics = "cannot use UR builtin kinematics with obstacles or interaction spaces"
+
+func (ua *URArm) useBuiltinKinematics(worldState *commonpb.WorldState, extra map[string]interface{}) (bool, error) {
+	var usingAtRuntime bool
+	if extra != nil {
+		if extraParam, ok := extra["builtin_kinematics"].(bool); ok {
+			usingAtRuntime = extraParam
+		}
+	}
+	if ua.builtinKinematics || usingAtRuntime {
```

Comment:
This isn't quite what we want, I think.

If a preference is explicitly specified on the gRPC call, that should always override whatever was specified at robot config time.

If nothing is specified, the default should be the robot config value.

Reply:
Something like this:
```
var useBuiltIn bool
if extra != nil {
	if extraParam, ok := extra["builtin_kinematics"].(bool); ok {
		useBuiltIn = extraParam
	} else {
		useBuiltIn = ua.builtinKinematics
	}
}

if useBuiltIn && worldState != nil && (len(worldState.Obstacles) != 0 || len(worldState.InteractionSpaces) != 0) {
	return false, errors.New(errBuiltinKinematics)
}

return useBuiltIn, nil
```